### ISSUE TITLE
Not enough info for fitToParent to calculate.

### ISSUE THREAD

**Issue by yohanes-ai** [funkhaus/fitToParent#4]
Using this code bellow
```
$('#iviFrame1').fitToParent();
$('#iviFrame2').fitToParent();
$(window).on('resize', function(){
	$('#iviFrame1').fitToParent();
	$('#iviFrame2').fitToParent();
}); 
```

Produce some error, here are the console log 
```
fitToParent.min.js:1 Uncaught Error: Not enough info for fitToParent to calculate.
    at r (fitToParent.min.js:1)
    at HTMLIFrameElement.<anonymous> (fitToParent.min.js:1)
    at Function.each (jquery-2.1.4.js:2)
    at n.fn.init.each (jquery-2.1.4.js:2)
    at n.fn.init.a.fn.fitToParent (fitToParent.min.js:1)
    at HTMLDocument.<anonymous> (?p=8&o={7f377407-04cc-464e-ac78-17738486e94a}&v=1.7.5.8:554)
    at j (jquery-2.1.4.js:2)
    at Object.fireWith [as resolveWith] (jquery-2.1.4.js:2)
    at Function.ready (jquery-2.1.4.js:2)
    at HTMLDocument.I (jquery-2.1.4.js:2)
```

Whats wrong?

**Comment by drewbaker** [comment]
You’ll get that error if the iframes are not in any parent elements I think. Do you have any HTML around the iframes? Can you provide a code pen by any chance?

**Comment by yohanes-ai** [comment]
@drewbaker sadly, the code it self are very complicated. We tried to implement your iFrame resizer in our Dashboard Portal. The Dashboard Portal contains two iFrame that can show some report and dashboard periodically and automatically switched. The application it self running on server side (Java Servlet) and Javascript generated by Server Side Script

Thus, here are the code
```

<main style="height:100%; width:100%;">        
        <iframe
            id="iviFrame1"
            name="iVi Dashboard"            
            longdesc="iVi - Your Personal Dashboard"
            style="display:inline"
            src=""
            width="100%"
            height="98%"
            align="middle"
            frameborder="0"
            marginwidth="0"
            marginheight="0"
            allowfullscreen="true"
            scrolling="auto"></iframe>
                
        <iframe
            id="iviFrame2"
            name="iVi Dashboard"
            longdesc="iVi - Your Personal Dashboard"
            style="display:none"
            src=""
            width="100%"
            height="98%"
            align="middle"
            frameborder="0"
            marginwidth="0"
            marginheight="0"
            allowfullscreen="true"
            scrolling="auto"></iframe>

        <div class="overlay">
            iVi - Your Personal Dashboard
        </div>
    </main>
</body>
```

[codepen](codepen.io/agungimannuel/post/fittoparent)

**Comment by drewbaker** [comment]
I can look into this more in the morning, but I suspect it’s because of the display none in the first iframe.

**Comment by johndigital** [comment]
@JhonImmanuel this was a bug with the way fitToParent was trying to read the dimensions off of the iframe attributes. The percentage value was throwing it off, it has now been fixed in e6109bbdd46f8cbf84baab8c40029c06dd19cbe8

Here's a jsFiddle with the new version, not throwing the error anymore: https://jsfiddle.net/bp5yb4aq/1/

Hope that helps!
